### PR TITLE
feat: Add examples to the spec JSON schema

### PR DESCRIPTION
**Pull request by candiduslynx**
part of https://github.com/cloudquery/cloudquery-issues/issues/1401

### CODE REVIEW

File: plugins/source/aws/client/spec/account.go [hunk 1]
```diff
@@ -3,10 +3,10 @@ package spec
 // This is used to specify one or more accounts to extract information from.
 type Account struct {
 	// Will be used as an alias in the source plugin and in the logs.
-	ID string `json:"id" jsonschema:"required,minLength=1"`
+	ID string `json:"id" jsonschema:"required,minLength=1" jsonschema:"example=some_id"`
```
Comment (erezrokah) [inline plugins/source/aws/client/spec/account.go:6]:
```suggestion
	ID string `json:"id" jsonschema:"required,minLength=1,example=some_id"`
```

File: plugins/source/aws/client/spec/account.go [hunk 2]
```diff
@@ -24,22 +24,22 @@ type Account struct {
...
-	RoleSessionName string `json:"role_session_name,omitempty"`
+	RoleSessionName string `json:"role_session_name,omitempty" jsonschema:"example=my_aws_role_session_name"`
 
 	// If specified will use this when assuming role to `role_arn`.
-	ExternalID string `json:"external_id,omitempty"`
+	ExternalID string `json:"external_id,omitempty" jsonschema:"example=external_id"`
 
 	// If specified, this region will be used as the default region for the account.
-	DefaultRegion string `json:"default_region,omitempty" jsonschema:"minLength=1,default=us-east-1"`
+	DefaultRegion string `json:"default_region,omitempty" jsonschema:"minLength=1,default=us-east-1,example=us-east-1"`
```
Comment (erezrokah) [inline plugins/source/aws/client/spec/account.go:39]: We don't need to add examples for fields that have a default, as the default serves as an example

File: plugins/source/aws/client/spec/spec.go [hunk 3]
```diff
@@ -22,44 +22,44 @@ type Spec struct {
...
 	// During initialization the AWS source plugin fetches information about each account and region.
 	// This setting controls how many accounts can be initialized concurrently.
 	// Only configurations with many accounts (either hardcoded or discovered via Organizations)
 	// should require modifying this setting, to either lower it to avoid rate limit errors, or to increase it to speed up the initialization process.
-	InitializationConcurrency int `json:"initialization_concurrency" jsonschema:"minimum=1,default=4"`
+	InitializationConcurrency int `json:"initialization_concurrency" jsonschema:"minimum=1,default=4,example=4"`
 
 	// The best effort maximum number of Go routines to use. Lower this number to reduce memory usage.
-	Concurrency int `json:"concurrency" jsonschema:"minimum=1,default=50000"`
+	Concurrency int `json:"concurrency" jsonschema:"minimum=1,default=50000,example=50000"`
```
Comment (candiduslynx) [inline plugins/source/aws/client/spec/spec.go:54]:
```suggestion
	Concurrency int `json:"concurrency" jsonschema:"minimum=1,default=50000"`
```

File: plugins/source/aws/client/spec/spec.go [hunk 4]
```diff
@@ -22,44 +22,44 @@ type Spec struct {
...
 
 	// When set to `true` plugin will sync data from APIs that incur a fee.
-	UsePaidAPIs bool `json:"use_paid_apis" jsonschema:"default=false"`
+	UsePaidAPIs bool `json:"use_paid_apis" jsonschema:"default=false,example=false"`
```
Comment (candiduslynx) [inline plugins/source/aws/client/spec/spec.go:57]:
```suggestion
	UsePaidAPIs bool `json:"use_paid_apis" jsonschema:"default=false"`
```

**Review by erezrokah — APPROVED** [review]

**Pull request merged**